backend/schedulerd/proxy_check.go

```diff
@@ -53,6 +53,7 @@ func substituteProxyEntityTokens(entity *corev2.Entity, check *corev2.CheckConfi
 	// entity
 	checkBytes, err := agent.TokenSubstitution(synthesizedEntity, check)
 	if err != nil {
+		logger.Error(fmt.Errorf("error in TokenSubstitution for proxy entity: %s check: %s err: %s", entity.Name, check.Name, err))
```

Comment:
You could do...

```go
logger.Errorf("error in TokenSubstitution for proxy entity: %s check: %s err: %s", entity.Name, check.Name, err)
```

but, given that logrus gives us structured logging I think it would be even more preferable to do:

```go
logger.WithField("check", check.Name).WithField("entity", entity.Name).WithError(err).Error("unable to substitute tokens")
```

Reply:
cool.. i'll refactor

Reply:
refactored now